client/landing/gutenboarding/stores/onboard/actions.ts

```diff
@@ -213,9 +211,8 @@ export const togglePageLayout = ( pageLayout: Template ) => ( {
 	pageLayout,
 } );
 
-export function* updatePlan( planSlug: Plans.PlanSlug ) {
-	const plan: Plans.Plan = yield select( PLANS_STORE, 'getPlanBySlug', planSlug );
-	yield setPlan( plan );
+export function* updatePlan( planProductId: number ) {
+	yield setPlanProductId( planProductId );
```

Comment:
Since this is now just setting a value, it doesn't need to be a generator function anymore. It can probably return directly the [action type](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-0250d6497eaaa43f73751fc190ae92a7d78e96be99f83f856ca3e100cd255d06R170-R173) `SET_PLAN_PRODUCT_ID`.